lib/OpenQA/Worker/Jobs.pm

```diff
@@ -164,6 +175,21 @@ sub upload {
 
     my $res = $OpenQA::Worker::Common::ua->start($tx);
 
+    # Upload known server failures (Instead of anything that's not 200)
+    unless ($res->res->code =~ m/^5/) {
```

Comment:
@okurz I prefer to use `unless` for readability: "Unless the code starts with a 5", keep going at least for this kind of checks, where I need some part of the code executed when a condition is met.